### PR TITLE
[ TreeView ] Multi Select state doesn't meet high contrast requirements

### PR DESCRIPTION
**Pull request by tashatitova**
Issue:

In high contrast the selected (checked) item was filled with high contrast color, the known issue with controls that have complex content - the background can't be flooded with Highlight color due to interaction states of content controls, here, Checkbox. On top of that, foreground was picking up states from ContentPresenter, resulting in incorrect high contrast:

multi-select

The fix is to only use borders to indicate hover and selected states going forward. We need to use two border outlines for MultiSelect case since the MultiSelect item selection is set independently of ContentPresenter visual states, so the hover appears on all items, selected or not. Easiest solution to this was to give MultiSelectGrid a border and a margin in high contrast so that inner outline fits within the outer outline.

https://user-images.githubusercontent.com/29714167/154332556-8a8fb20c-111b-4718-827e-23eaa028766f.mp4

Updating focus visuals while at it (overlapped with some of the nested borders):

focus-updated

Multi-select high contrast, edge case scenario, selected item + hover + focus:
focus-hc-=updated

### CODE REVIEW

File: dev/TreeView/TreeView_themeresources.xaml [hunk 1]
```diff
@@ -33,7 +33,8 @@
             <StaticResource x:Key="TreeViewItemBorderBrushSelectedPointerOver" ResourceKey="SubtleFillColorTransparentBrush" />
             <StaticResource x:Key="TreeViewItemBorderBrushSelectedPressed" ResourceKey="SubtleFillColorTransparentBrush" />
             <StaticResource x:Key="TreeViewItemBorderBrushSelectedDisabled" ResourceKey="SubtleFillColorTransparentBrush" />
-            
+
+            <StaticResource x:Key="TreeViewItemBorderBrushMultiSelectSelected" ResourceKey="SubtleFillColorTransparentBrush" />
```
Comment (karenbtlai) [inline dev/TreeView/TreeView_themeresources.xaml:37]: Can this resource name be simplified or shortened?

Comment (tashatitova) [reply dev/TreeView/TreeView_themeresources.xaml:37]: @karenbtlai I don't think we can drop anything from here, as just 'selected' applies to a regular selected tree view item (different visual state). I agree this one is long but I wanted it to be descriptive, not confusing

**Review by marksfoster — APPROVED** [review]

File: dev/TreeView/TreeView_themeresources.xaml (hunk 1 above)
Comment (karenbtlai) [reply dev/TreeView/TreeView_themeresources.xaml:37]: Let's request feedback from more people as once it's added it cannot be removed. I think we can shorten it to simply *MultiSelect as that implies it being selected. @StephenLPeters @gabbybilka thoughts?

Comment (StephenLPeters) [reply dev/TreeView/TreeView_themeresources.xaml:37]: I don't think you can shorten that way, MultiSelect is talking about the state of the treeview, not the item.  You could maybe do TreeViewItemBorderBrushMultiSelected, but to me that feels a little bit hidden.

**Review by StephenLPeters — APPROVED** [review]

**Pull request merged**